src/components/ArticlesByTopic.jsx

```diff
@@ -0,0 +1,27 @@
+import { useParams } from "react-router-dom";
+import { ArticleCard } from "./ArticleCard";
+
+
+
+export const ArticlesByTopic = ({articles}) => {
+    const {topic} = useParams();
+
+
+    const filteredArticles = articles.filter((article) => {
```

Comment:
You don't want to be doing FE filtering as this could lead to stagnant data. If you want all the articles from a specific topic make a request to the API